component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAuthenticator.java

```diff
@@ -519,7 +517,7 @@ private boolean isInitialFederationAttempt(AuthenticationContext context) {
      * @throws BackupCodeException If an error occurred while validating token.
      */
     private boolean isValidTokenFederatedUser(String code, AuthenticationContext context, String userName)
-            throws BackupCodeException {
+            throws BackupCodeException, NoSuchAlgorithmException {
```

Comment:
We should not throw this. Shall we wrap this with backup code server exceptions?

Reply:
addressed with [7d9674b](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/7d9674b6bf172b43c30479b0d64fdfa6f8130373)